torch/autograd/_functions/reduce.py

```diff
@@ -46,13 +46,44 @@ def forward(self, input):
     def backward(self, grad_output):
         if self.dim is None:
             input, = self.saved_tensors
-            grad_input = grad_output.new(self.input_size).fill_(self.result)
-            return grad_input.div(input)
+            zero_loc = (input==0).nonzero()
+            if zero_loc.dim() == 0:
+                grad_input = grad_output.new(self.input_size).fill_(self.result)
+                return grad_input.div(input)
+            elif zero_loc.size()[0] > 1:
+                return grad_output.new(self.input_size).fill_(0)
+            else:
+                grad_input = grad_output.new(self.input_size).fill_(0)
+                indexing_tuple = tuple(zero_loc[0].numpy())
```

Comment:
Yes you are right.

This will be fixed as
indexing_tuple = tuple((zero_loc[0].cpu() if zero_loc.is_cuda else zero_loc[0]).numpy())
after receiving other feedbacks.

Thanks!